pkg/apis/internal.admin.acorn.io/v1/resources.go

```diff
@@ -30,44 +37,78 @@ type Resources struct {
 
 // Add will add the resources of another Resources struct into the current one.
 func (current *Resources) Add(incoming Resources) {
-	current.Apps += incoming.Apps
-	current.Containers += incoming.Containers
-	current.Jobs += incoming.Jobs
-	current.Volumes += incoming.Volumes
-	current.Secrets += incoming.Secrets
-	current.Images += incoming.Images
-	current.Projects += incoming.Projects
-
-	current.VolumeStorage.Add(incoming.VolumeStorage)
-	current.Memory.Add(incoming.Memory)
-	current.CPU.Add(incoming.CPU)
+	add := func(c, i int) int {
+		if c == Unlimited || i == Unlimited {
+			return Unlimited
+		}
+		return c + i
+	}
+
+	addQuantity := func(c, i resource.Quantity) resource.Quantity {
+		if c.Equal(UnlimitedQuantity()) || i.Equal(UnlimitedQuantity()) {
+			return UnlimitedQuantity()
+		}
+		c.Add(i)
+		return c
+	}
+
+	current.Apps = add(current.Apps, incoming.Apps)
+	current.Containers = add(current.Containers, incoming.Jobs)
+	current.Jobs = add(current.Jobs, incoming.Jobs)
+	current.Volumes = add(current.Volumes, incoming.Volumes)
+	current.Secrets = add(current.Secrets, incoming.Secrets)
+	current.Images = add(current.Images, incoming.Images)
+	current.Projects = add(current.Projects, incoming.Projects)
+
+	current.VolumeStorage = addQuantity(current.VolumeStorage, incoming.VolumeStorage)
+	current.Memory = addQuantity(current.Memory, incoming.Memory)
+	current.CPU = addQuantity(current.CPU, incoming.CPU)
 }
 
 // Remove will remove the resources of another Resources struct from the current one.
 func (current *Resources) Remove(incoming Resources, all bool) {
-	// Do not allow resources to go below 0
-	nonNegativeSubtract := func(currentVal, incomingVal int) int {
-		difference := currentVal - incomingVal
+	sub := func(c, i int) int {
+		if c == Unlimited && i == Unlimited {
+			return 0
+		} else if c == Unlimited || i == Unlimited {
+			return c
+		}
```

Comment:
I think your current code has a bug if `c` is some value and `i` is `Unlimited`.

Cases:
1. `c unlimited i unlimited -> 0`
1. `c unlimited i value -> unlimited`
1. `c value i unlimited -> 0`
1. `c value i value -> fallthrough case`

Here's the logic I came up with:
```
if i == Unlimited {
   return 0
} else if c == Unlimited {
   return c
}
```